### DISCUSSION TITLE
How to read files from a ghidra extension?

### DISCUSSION THREAD

**Discussion by aldelaro5** [NationalSecurityAgency/ghidra#3875] [Q&A] [answered]
I have been trying everything and I am tearing my hair out trying to figure out how to do this so I'll just ask.

How do you read a simple text file from a ghidra module assuming you ship the file with your extension?

See, I can put the file in the data directory and if I do that, everything works in Eclipse if the path of the file is `data/file` where file is the actual filename. Once I export the extension, the file is in the zip in the data directory of the extension so everything looks good.

But if I actually install the zip in ghidra, now it can't find the file. Since I have no idea how you get logs from a loaded extension (system.out.println doesn't work in ghidra, only in Eclipse), I can't know what path it is in so I can't know how you are supposed to load the file.

How can I make it work in BOTH Eclipse AND ghidra?

**Comment by ryanmkurtz** [comment, marked as answer]
There are a bunch of methods in `Application.java` for reading files relative to where your module lives.  They start with `getModule*`.

**Reply by aldelaro5** [reply]
That did it, thanks.
